### PR TITLE
Add benchmarks for merge package

### PR DESCRIPTION
**Pull request by apelisse**
This is testing the high-level part of SMD, we would probably need
lower-level benchmarks in this repo, but this at least gives us a
general idea of the direction.

/assign @lavalamp @jennybuckley

### CODE REVIEW

File: merge/deduced_test.go [hunk 1]
```diff
@@ -534,3 +534,280 @@ func TestDeduced(t *testing.T) {
...
+						    f:
+						    - name: n
+						      value: 1
+					`,
+			},
+		},
+		Object: `
+				a: 2
+				b:
+				  c:
```
Comment (lavalamp) [inline merge/deduced_test.go]: mixed tabs and spaces?

Comment (apelisse) [reply merge/deduced_test.go]: Actually, gofmt moved `Object` one level down when I moved TestCase outside the list, but obviously it doesn't indent yaml in strings...

Comment (apelisse) [reply merge/deduced_test.go]: OK I fixed this.

**Pull request merged**